src/mdx-components.tsx

```diff
@@ -14,7 +15,12 @@ export function useMDXComponents(components: MDXComponents): MDXComponents {
     //   <h1 className="text-2xl font-semibold md:text-6xl">{children}</h1>
     // ),
     h1: ({ children }) => (
-      <h1 className="text-3xl font-semibold  md:text-5xl">{children}</h1>
+      <h1
+        className="text-3xl font-semibold  md:text-5xl"
+        id={slugifyTitle(children as string)}
+      >
+        {children}
+      </h1>
```

Comment:
There is a plugin that can take care of this, see:
* https://github.com/rehypejs/rehype-slug
* https://github.com/rehypejs/rehype-autolink-headings

I am perfectly fine with this solution as it would be a temporary one, but I'll leave it to you if you want to use a plugin for the markdown and style the hash with CSS using a pseudo-selector `:after`. The headings that aren't in markdown could easily be matched with this.